metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -804,7 +817,7 @@ void handleBrokerUnfenced(int brokerId, long brokerEpoch, List<ApiMessageAndVers
                 TopicControlInfo topic = topics.get(topicEntry.getValue());
                 if (topic != null) {
                     for (int partitionId : topic.parts.keySet()) {
-                        ApiError error = electLeader(topicName, partitionId, uncleanOk, records);
+                        ApiError error = electLeader(topicName, partitionId, electionType, records);
```

Comment:
Good point. I've updated the code to handle this case. It is covered in the unit tests, so perhaps that is good enough? It would be difficult to test through `LeaderElectionCommandTest`, so we would need an `ElectLeadersRequestTest` if we want an integration test.

Reply:
Okay, sounds good to me. We have `PlaintextAdminIntegrationTest` but that suite is tightly coupled to the ZK implementation.